Remove trailers from store files since they are redundant

davidegrohmann opened the pull request:
They are _not_ used to determine recovery and they are _not_ useful for rebuilding id generators since ".id" files have their own way to check if they were cleanly closed or not.

This commit removes the trailers from the stores.

Note: the counts store relies on  the trailer to determine if the latest version was completely flushed on disk or not.  In order to preserve this check now we write as trailer the "format specifier" which can be found in the header as well.

For future reference: Trailers are used before migration to check that the stores are closed cleanly, since we have trailers in older store previous 2.3, this is fine.  For post and 2.3 stores the best way of
implementing this would be to check that the latest entry in the transaction log is a checkpoint.

chrisvest commented:
There are a lot of formatting changes that are unrelated to store trailers. It would have been nice if those changes had been put in a separate commit.

davidegrohmann commented:
@chrisvest true, I have messed up in having small commits on this branch. I can try to cleanup if you think it is important.

chrisvest commented:
@davidegrohmann As it is now, Github can't display the whole diff, so yeah, it would be very helpful.

davidegrohmann committed:
Remove trailers from store files since they are redundant

They are _not_ used to determine recovery and they are _not_ useful
for rebuilding id generators since ".id" files have their own way to
check if they were cleanly closed or not.

This commit removes the trailers from the stores.

Note: the counts store relies on the trailer to determine if the
latest version was completely flushed on disk or not.  In order to
preserve this check now we write as trailer the "format specifier"
which can be found in the header as well.

For future reference: Trailers are used before migration to check that
the stores are closed cleanly, since we have trailers in older store
previous 2.3, this is fine.  For post and 2.3 stores the best way of
implementing this would be to check that the latest entry in the
transaction log is a checkpoint.

davidegrohmann commented:
@chrisvest cleanup done. Now the diff should be easier to understand.

MishaDemianenko commented on community/kernel/src/main/java/org/neo4j/kernel/impl/store/DynamicArrayStore.java:
```diff
@@ -56,23 +53,20 @@ public DynamicArrayStore(
-    public String getTypeDescriptor()
+    public String getStoreName()
     {
         return TYPE_DESCRIPTOR;
```
constant should be renamed as well here

MishaDemianenko commented on community/kernel/src/main/java/org/neo4j/kernel/impl/store/DynamicStringStore.java:
```diff
@@ -62,7 +55,7 @@ public DynamicStringStore(
-    public String getTypeDescriptor()
+    protected String getStoreName()
     {
         return TYPE_DESCRIPTOR;
```
also constant renaming, probably comment valid for all the stores

MishaDemianenko commented on community/kernel/src/main/java/org/neo4j/kernel/impl/store/StoreFactory.java:
```diff
@@ -82,54 +80,33 @@
     {
-        this( storeDir, config, idGeneratorFactory, pageCache, fileSystemAbstraction, logProvider,
-                monitors, StoreVersionMismatchHandler.FORCE_CURRENT_VERSION );
+        this( null, config, idGeneratorFactory, pageCache, fileSystemAbstraction, logProvider );
```
looks weird that first constructor parameter is null

lutovich commented on community/kernel/src/test/java/org/neo4j/kernel/impl/storemigration/legacystore/LegacyStoreVersionCheckTest.java:
```diff
@@ -0,0 +1,104 @@
+    private File fileContaining( EphemeralFileSystemAbstraction fs, String content ) throws IOException
+    {
+        File shortFile = new File( "shortFile" );
+        OutputStream outputStream = fs.openAsOutputStream( shortFile, true );
```
Maybe try-with-resources?

MishaDemianenko commented on community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/StoreUpgrader.java:
```diff
@@ -150,32 +151,42 @@ public void migrateIfNeeded( File storeDirectory, SchemaIndexProvider schemaInde
-            // We don't need to migrate if we're at the phase where we have migrated successfully
-            // and it's just a matter of moving over the files to the storeDir.
-            if ( !migrationStatusIs( migrationStateFile, MigrationStatus.moving ) )
+        if ( migrationStatus == null || migrationStatus.ordinal() <= MigrationStatus.moving.ordinal() )
```
Handling of null values in `migrationStatus` and `versionToMigrateFrom` is overcomplicated and tricky.
WDYT about updating migrationStatus to current status in each step instead and just read `versionToMigrateFrom` since that will be always read except we are in `completed` phase that should not be the case here

chrisvest commented:
@davidegrohmann This needs a rebase. It looked good when I reviewed it. We agreed to wait with merging until after the next milestone cut.

tinwelint commented:
@davidegrohmann @chrisvest so this should be good to merge now I'd guess
